my/core/query.py

```diff
@@ -206,6 +238,18 @@ def _drop_unsorted(itr: Iterator[ET], orderfunc: OrderFunc) -> Iterator[ET]:
         yield o
 
 
+# try getting the first value from the iterator
+# similar to my.core.common.warn_if_empty? this doesnt go through the whole iterator though
```

Comment:
oh good point, maybe I forgot about more_itetrools back then
although it's injected in the iterable itself, and the only way to use the iterable is to iterate all of it.. so I guess the usecase is a bit different